### PR TITLE
✨amp-ima-video: Show Controls on Hover

### PR DESCRIPTION
**Pull request by torch2424**
relates to #19393

This enables showing ima-video controls on hover, instead of onclick. This also makes sure to work on mobile devices to preserve the tap to show controls behavior, as it is in amp-video.

From a UX perspective, got an approval by @spacedino to make `amp-ima-video` to match the behavior of [`amp-video`](https://ampbyexample.com/components/amp-video/) as much as possible, and this PR falls inline with that 😄.

cc @curseagain 

# Example

## Desktop

imavideohover

## Mobile

imavideomobile

### CODE REVIEW

File: ads/google/imaVideo.js [hunk 1]
```diff
@@ -579,7 +607,7 @@ function onImaLoadSuccess(global, data) {
 function onImaLoadFail() {
   // Something blocked ima3.js from loading - ignore all IMA stuff and just play
   // content.
-  videoPlayer.addEventListener(interactEvent, showControls);
+  addHoverEventToElement(/** @type !Element*/(videoPlayer), showControls);
```
Comment (zhouyx) [inline ads/google/imaVideo.js]: nit: extra space after `*/`

File: ads/google/imaVideo.js [hunk 2]
```diff
@@ -1278,7 +1306,11 @@ export function resetControlsAfterAd() {
...
  */
 export function showControls() {
-  setStyle(controlsDiv, 'display', 'flex');
+  if (!controlsVisible) {
+    setStyle(controlsDiv, 'display', 'flex');
+    controlsVisible = true;
+  }
+
   // Hide controls after 3 seconds
   if (playerState == PlayerStates.PLAYING) {
```
Comment (zhouyx) [inline ads/google/imaVideo.js]: I was playing with the `mousemove` event, and it seems that it fires a LOT. I am not sure about the cost of `setTimeout` and `clearTimeout`. Do you think it's fine w/o a throttle?

Comment (torch2424) [reply ads/google/imaVideo.js]: Yeah the `mousemove` event does fire a lot, but I was thinking about it and I essentially came to two conclusions:

1. The event will only be fired on desktop devices, during certain sections of watching the video.
2. Adding a throttle *may* improve user experience, however, I couldn't think of a good way to implement this that would be flexible and testable, thus hurting developer experience.

Though, I do think this is worth having a quick chat about on how we can easily add/remove this dynamic event, that has a throttle. 😄 I'll reach out tommorow, or vice/versa 👍

**Comment by torch2424** [comment]
@zhouyx @curseagain made all requested changes! Added a throttler as well 😄

File: ads/google/imaVideo.js [hunk 3]
```diff
@@ -507,6 +519,35 @@ export function imaVideo(global, data) {
...
 
+/**
+ * Adds the appropriate event listener to an element
+ * to represent a hover state.
+ *
+ * NOTE: This does not add a throttler,
+ * since this is applied per element,
+ * and would require wrapping the callback.
+ * Thus, the callback passed should be,
+ * appropriately throttled. See showControlsThrottled.
```
[the remark this answers is not in the document]
Comment (torch2424) [reply ads/google/imaVideo.js]: Thanks! 😄

**Comment by torch2424** [comment]
@zhouyx This is good to go! Fixed the rate-limit build system thing 😄

**Comment by torch2424** [comment]
@zhouyx No worries! Thank you!

**Pull request merged**